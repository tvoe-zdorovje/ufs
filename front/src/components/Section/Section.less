@import "../Shared";

.section {
  margin-top: 12px;
  border: 1px solid @green;
}

.header {
  padding: 1px 8px 1px 4px;
  margin: -12px 0 0 8px;
  color: @green;
  background-color: #fff;
  position: absolute;
  z-index: 10;
}

.container {
  padding: 10px 5px 5px 5px;
}
