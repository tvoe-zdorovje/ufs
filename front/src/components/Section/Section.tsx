import React from 'react'

const styles = require('./Section.less')

interface SectionProps {
  header: string,
  children?: any,
}

export default (props: SectionProps) => (
  <section className={styles.section}>
    <div className={styles.header}>{props.header}</div>
    <div className={styles.container}>
      {props.children}
    </div>
  </section>
);
