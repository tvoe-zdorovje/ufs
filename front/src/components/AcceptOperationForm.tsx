import React from 'react'
import {
  Alert,
  Button,
  Col,
  Input,
  Form,
  Row
} from 'antd'
import Section from './Section/Section'
import m2w from 'utils/MoneyToWords'

const styles = require('./Shared.less')

const FormItem = Form.Item

const formItemLayout = {
  labelCol: { span: 4 },
  wrapperCol: { span: 16 },
}

export interface AcceptOperationFormProps {
  completeOperation: Function,
  onFinish: Function,
  amount: number,
  confirmed: boolean,
  complete: boolean,
  workplaceId: string,
  taskId: string,
  packageId: string,
  operationTypeCode: string,
  operationId: string,
  loadingConfirm: boolean,
  loadingCancel: boolean,
}

export default class AcceptOperationForm extends React.Component<AcceptOperationFormProps, any> {
  complete = (confirm) => {
    const { packageId, taskId, workplaceId, operationTypeCode } = this.props
    this.props.completeOperation({ packageId, taskId, workplaceId, operationTypeCode }, confirm)
  }

  renderIncomplete = (amount: number, loadingConfirm: boolean, loadingCancel: boolean) => (
    <Col span={16}>
      <Section header="Прием наличности">
        <Row>
          <FormItem className={styles.formItem} label="Сумма к приему" {...formItemLayout}>
            <Input size="large" disabled value={`${amount},00`} className={styles.bold} />
          </FormItem>
          <FormItem className={styles.formItem} label="Сумма прописью" {...formItemLayout}>
            <Input size="large" disabled value={m2w(amount)} className={styles.bold} />
          </FormItem>
        </Row>
      </Section>
      <Row className={styles.rightAligned}>
        <Button
          className={styles.greenButton}
          loading={loadingConfirm}
          onClick={() => this.complete(true)}
        >
          Завершить операцию
        </Button>
        <Button
          className={styles.redButton}
          loading={loadingCancel}
          onClick={() => this.complete(false)}>
          Отменить операцию
        </Button>
      </Row>
    </Col>
  );

  renderComplete = () => (
    <Col span={16}>
      <Section header="Прием наличности">
        <Alert
          message={`Операция успешно ${this.props.confirmed ? 'завершена' : 'отменена'}`}
          description="Вы можете вернуться в главное меню"
          type="success"
        />
      </Section>
      <Row>
        <Button className={styles.greenButton} onClick={() => this.props.onFinish()}>
          Вернуться в главное меню
        </Button>
      </Row>
    </Col>
  )

  render() {
    const {amount, loadingConfirm, loadingCancel} = this.props
    return (this.props.complete ? this.renderComplete() : this.renderIncomplete(amount, loadingConfirm, loadingCancel))
  }
}
