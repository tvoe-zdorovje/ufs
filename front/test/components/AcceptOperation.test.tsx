import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import AcceptOperationForm, { AcceptOperationFormProps } from 'components/AcceptOperationForm'
import Section from 'components/Section/Section'
import { Alert } from 'antd'

describe('Компонент AcceptOperationForm', () => {
  const props: AcceptOperationFormProps = {
    completeOperation: () => {
    },
    onFinish: () => {
    },
    amount: 100,
    confirmed: false,
    complete: false,
    workplaceId: '',
    taskId: '',
    packageId: '',
    operationTypeCode: '',
    operationId: '',
    loadingConfirm: true,
    loadingCancel: true,
  }
  const shallowWrapper = shallow(<AcceptOperationForm {...props} />)
  const renderWrapper = render(<AcceptOperationForm {...props} />)

  it('отображает форму для завершенной операции', () => {
    shallowWrapper.setProps({ complete: true })
    shallowWrapper.update()
    expect(shallowWrapper.find(Alert)).toHaveLength(1)
  })

  it('использует свойства <AcceptOperationForm />', () => {
    expect(shallowWrapper.find(Section).prop('header')).toBe('Прием наличности')
  })

  it('отображает компонент <Section />', () => {
    expect(shallowWrapper.find(Section)).toHaveLength(1)
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
