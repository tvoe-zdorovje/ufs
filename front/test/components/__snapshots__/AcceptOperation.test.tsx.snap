// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент AcceptOperationForm совпадает со своим снимком 1`] = `
<div
  class="ant-col-16"
>
  <section
    class="section"
  >
    <div
      class="header"
    >
      Прием наличности
    </div>
    <div
      class="container"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-row ant-form-item formItem"
        >
          <div
            class="ant-col-4 ant-form-item-label"
          >
            <label
              class=""
              title="Сумма к приему"
            >
              Сумма к приему
            </label>
          </div>
          <div
            class="ant-col-16 ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <input
                class="ant-input ant-input-lg bold"
                disabled=""
                type="text"
                value="100,00"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item formItem"
        >
          <div
            class="ant-col-4 ant-form-item-label"
          >
            <label
              class=""
              title="Сумма прописью"
            >
              Сумма прописью
            </label>
          </div>
          <div
            class="ant-col-16 ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <input
                class="ant-input ant-input-lg bold"
                disabled=""
                type="text"
                value="сто рублей 00 коп."
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="ant-row rightAligned"
  >
    <button
      class="ant-btn ant-btn-loading greenButton"
      type="button"
    >
      <i
        class="anticon anticon-spin anticon-loading"
      />
      <span>
        Завершить операцию
      </span>
    </button>
    <button
      class="ant-btn ant-btn-loading redButton"
      type="button"
    >
      <i
        class="anticon anticon-spin anticon-loading"
      />
      <span>
        Отменить операцию
      </span>
    </button>
  </div>
</div>
`;
